### PR TITLE
tracing: trace capability checks performed by the kernel on X access

### PR DESCRIPTION
**Pull request by tixxdz**
Allow to trace capability checks performed by the kernel on processes. 

It adds both the capability and user namespace types, which makes it easy to see if the requested capability is within a host or a private user namespace (which can be setup by unprivileged users on some distros), also it will print capability in value and name.

Example:
```
  "process_kprobe": {
    "process": {
      "exec_id": "OjEyMjY0NTMxODI1ODkzOjExNDI0Ng==",
      "pid": 114246,
      "uid": 0,
      "cwd": "/root",
      "binary": "/usr/bin/head",
      "arguments": "-c 1 /proc/kcore",
      "flags": "execve clone",
      "start_time": "2022-09-26T15:59:01.548066118Z",
      "auid": 1000,
      "parent_exec_id": "OjEyMjMxMjg5NjkzNDQxOjExNDE3NQ==",
      "refcnt": 1
    },
    "function_name": "cap_capable",
    "args": [
      {
        "cred_arg": {
          "permitted": [
            "CAP_CHOWN",
            "DAC_OVERRIDE",
            "CAP_DAC_READ_SEARCH",
            "CAP_FOWNER",
            "CAP_FSETID",
            "CAP_KILL",
            "CAP_SETGID",
            "CAP_SETUID",
            "CAP_SETPCAP",
            "CAP_LINUX_IMMUTABLE",
            "CAP_NET_BIND_SERVICE",
            "CAP_NET_BROADCAST",
            "CAP_NET_ADMIN",
            "CAP_NET_RAW",
            "CAP_IPC_LOCK",
            "CAP_IPC_OWNER",
            "CAP_SYS_MODULE",
            "CAP_SYS_RAWIO",
            "CAP_SYS_CHROOT",
            "CAP_SYS_PTRACE",
            "CAP_SYS_PACCT",
            "CAP_SYS_ADMIN",
            "CAP_SYS_BOOT",
            "CAP_SYS_NICE",
            "CAP_SYS_RESOURCE",
            "CAP_SYS_TIME",
            "CAP_SYS_TTY_CONFIG",
            "CAP_MKNOD",
            "CAP_LEASE",
            "CAP_AUDIT_WRITE",
            "CAP_AUDIT_CONTROL",
            "CAP_SETFCAP",
            "CAP_MAC_OVERRIDE",
            "CAP_MAC_ADMIN",
            "CAP_SYSLOG",
            "CAP_WAKE_ALARM",
            "CAP_BLOCK_SUSPEND",
            "CAP_AUDIT_READ",
            "CAP_PERFMON",
           "CAP_BPF",
            "CAP_CHECKPOINT_RESTORE"
          ],
          "effective": [
            "CAP_CHOWN",
            "DAC_OVERRIDE",
            "CAP_DAC_READ_SEARCH",
            "CAP_FOWNER",
            "CAP_FSETID",
            "CAP_KILL",
            "CAP_SETGID",
            "CAP_SETUID",
            "CAP_SETPCAP",
            "CAP_LINUX_IMMUTABLE",
            "CAP_NET_BIND_SERVICE",
            "CAP_NET_BROADCAST",
            "CAP_NET_ADMIN",
            "CAP_NET_RAW",
            "CAP_IPC_LOCK",
            "CAP_IPC_OWNER",
            "CAP_SYS_MODULE",
            "CAP_SYS_RAWIO",
            "CAP_SYS_CHROOT",
            "CAP_SYS_PTRACE",
            "CAP_SYS_PACCT",
            "CAP_SYS_ADMIN",
            "CAP_SYS_BOOT",
            "CAP_SYS_NICE",
            "CAP_SYS_RESOURCE",
            "CAP_SYS_TIME",
            "CAP_SYS_TTY_CONFIG",
            "CAP_MKNOD",
            "CAP_LEASE",
            "CAP_AUDIT_WRITE",
            "CAP_AUDIT_CONTROL",
            "CAP_SETFCAP",
            "CAP_MAC_OVERRIDE",
            "CAP_MAC_ADMIN",
            "CAP_SYSLOG",
            "CAP_WAKE_ALARM",
            "CAP_BLOCK_SUSPEND",
            "CAP_AUDIT_READ",
            "CAP_PERFMON",
            "CAP_BPF",
            "CAP_CHECKPOINT_RESTORE"
          ]
        }
      },
     {
        "user_namespace_arg": {
          "owner": 0,
          "group": 0,
          "ns": {
            "inum": 4026531837,
            "is_host": true
          }
        }
      },
      {
        "capability_arg": {
          "value": 17,
          "name": "CAP_SYS_RAWIO"
        }
      }
    ],
    "return": {
      "int_arg": 0
    },
    "action": "KPROBE_ACTION_POST"
```

Signed-off-by: Djalal Harouni <tixxdz@gmail.com>

### CODE REVIEW

**Review by kkourt — APPROVED** [review]
LGTM! @tpapagian Could you also have a look please?

**Comment by tixxdz** [comment]
Hold on this one

File: pkg/sensors/tracing/kprobe_test.go [hunk 1]
```diff
@@ -2389,6 +2389,75 @@ spec:
...
+	obs, err := observer.GetDefaultObserverWithFile(t, ctx, testConfigFile, tus.Conf().TetragonLib)
+	if err != nil {
+		t.Fatalf("GetDefaultObserverWithFile error: %s", err)
+	}
+	observer.LoopEvents(ctx, t, &doneWG, &readyWG, obs)
+	readyWG.Wait()
+
+	kpChecker := ec.NewProcessKprobeChecker().
+		WithFunctionName(sm.Full("cap_capable")).
+		WithAction(tetragon.KprobeAction_KPROBE_ACTION_POST)
```
Comment (tpapagian) [inline pkg/sensors/tracing/kprobe_test.go:2440]: It should be good if we can check more here. For example, based on the comment below we expect "capability" to be `CAP_SYS_RAWIO`. Maybe we can also check the user namespace value (we have some helpers in `pkg/reader/namespace/namespace.go`).

**Review by tpapagian — APPROVED** [review]
LGTM, thanks! 

I have a comment for testing but this can be also done as a follow-up.

**Review by willfindlay — APPROVED** [review]

File: pkg/sensors/tracing/kprobe_test.go (hunk 1 above)
Comment (tixxdz) [reply pkg/sensors/tracing/kprobe_test.go]: Fixed! thank you @tpapagian

**Pull request merged**